File: problem-solving/baekjoon/10546_배부른_마라토너.go
// https://www.acmicpc.net/problem/10546
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)

	var runners = map[string]int{}
	for i := 0; i < n; i++ {
		var runner string
		fmt.Fscanln(reader, &runner)
		runners[runner]++
	}
	for i := 0; i < n-1; i++ {
		var finished string
		fmt.Fscanln(reader, &finished)
		runners[finished]--
		if runners[finished] == 0 {
			delete(runners, finished)
		}
	}
	for key := range runners {
		fmt.Fprintln(writer, key)
	}
}

File: problem-solving/baekjoon/10546_배부른_마라토너.py
# https://www.acmicpc.net/problem/10546
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    runners = {}
    for i in range(n):
        runner = sys.stdin.readline().rstrip()
        if runner in runners:
            runners[runner] += 1
        else:
            runners[runner] = 1
    for i in range(n-1):
        finished = sys.stdin.readline().rstrip()
        if runners[finished] == 1:
            runners.pop(finished)
        else:
            runners[finished] -= 1
    for i in runners.keys():
        print(i)